custom_components/candy/binary_sensor.py

```diff
@@ -0,0 +1,325 @@
+    def device_name(self) -> str:
+        return DEVICE_NAME_HOB
+
+    def suggested_area(self) -> str:
+        return SUGGESTED_AREA_KITCHEN
+
+    @property
+    def name(self) -> str:
+        return self.device_name()
+
+    @property
+    def unique_id(self) -> str:
+        return UNIQUE_ID_HOB_BINARY.format(self.config_id)
+
+    @property
+    def icon(self) -> str:
+        return "mdi:pot-steam"
+
+    @property
+    def available(self):
+        status: HobStatus = self.coordinator.data
+        return status is not None
+
+    @property
+    def is_on(self):
+        status: HobStatus = self.coordinator.data
+        return status.machine_state not in [HobState.IDLE] 
+
+class CandyDishwasherBinarySensor(CandyBaseBinarySensor):
+    def device_name(self) -> str:
+        return DEVICE_NAME_DISHWASHER
+
+    def suggested_area(self) -> str:
+        return SUGGESTED_AREA_KITCHEN
+
+    @property
+    def name(self) -> str:
+        return self.device_name()
+
+    @property
+    def unique_id(self) -> str:
+        return UNIQUE_ID_DISHWASHER_BINARY.format(self.config_id)
+
+    @property
+    def icon(self) -> str:
+        return "mdi:glass-wine"
+
+    @property
+    def available(self):
+        status: DishwasherStatus = self.coordinator.data
+        return status is not None
+
+    @property
+    def is_on(self):
+        status: DishwasherStatus = self.coordinator.data
+        return status.machine_state not in [DishwasherState.IDLE] 
+
+class CandyHoodBinarySensor(CandyBaseBinarySensor):
+    def device_name(self) -> str:
+        return DEVICE_NAME_HOOD
+
+    def suggested_area(self) -> str:
+        return SUGGESTED_AREA_KITCHEN
+
+    @property
+    def name(self) -> str:
+        return self.device_name()
+
+    @property
+    def unique_id(self) -> str:
+        return UNIQUE_ID_HOOD_BINARY.format(self.config_id)
+
+    @property
+    def icon(self) -> str:
+        return "mdi:fan"
+
+    @property
+    def available(self):
+        status: HoodStatus = self.coordinator.data
+        return status is not None
+
+    @property
+    def is_on(self):
+        status: HoodStatus = self.coordinator.data
+        return status.machine_state not in [HoodState.IDLE] 
+
+class CandyFridgeBinarySensor(CandyBaseBinarySensor):
+    def device_name(self) -> str:
+        return DEVICE_NAME_FRIDGE
+
+    def suggested_area(self) -> str:
+        return SUGGESTED_AREA_KITCHEN
+
+    @property
+    def name(self) -> str:
+        return self.device_name()
+
+    @property
+    def unique_id(self) -> str:
+        return UNIQUE_ID_FRIDGE.format(self.config_id)
+
+    @property
+    def icon(self) -> str:
+        return "mdi:fridge"
+
+    @property
+    def available(self):
+        status: FridgeStatus = self.coordinator.data
+        return status is not None
+
+    @property
+    def is_on(self):
+        status: FridgeStatus = self.coordinator.data
+        return status is not None
```

Comment:
What is the purpose of this bianry sensor if it's always on (except for errors)?

Reply:
Yes it is true is actually allways on only when user disconnect from power socket. 
But actually I put this code that have all appliances binary sensor which tell us if device work or not in generally.

Reply:
Let's add only the binary sensors that make sense, it doesn't have to have 1 for each supported appliance.